### DISCUSSION TITLE
Order of inherited classes in PooledJSONRPCServer definition

### DISCUSSION THREAD

**Discussion by RobiIhs** [tcalmant/jsonrpclib#54] [Ideas]
Hello,
first of all, thanks for the library.

I tried to use _PooledJSONRPCServer_ but it didn't seem to work as expected.
Could it be because _socketserver.ThreadingMixIn_ should precede _SimpleJSONRPCServer_ in inheritance order?

Then I created my own _ThreadedJSONRPCServer_ inheriting from _SimpleJSONRPCServer_ and _socketserver.ThreadingMixIn_ and I verified that  _socketserver.ThreadingMixIn_ must come first to have multi-threaded behaviour.

Roberto

cfr.:
https://docs.python.org/3/library/socketserver.html#:~:text=The%20mix%2Din%20class%20comes%20first%2C%20since%20it%20overrides%20a%20method%20defined%20in%20UDPServer.%20Setting%20the%20various%20attributes%20also%20changes%20the%20behavior%20of%20the%20underlying%20server%20mechanism.

**Comment by tcalmant** [comment]
Hi,

Indeed it seems there is an issue there, as a MixIn should always be before the base class.
I'll take a deeper look this weekend, especially trying to see what test could be written to check that issue.

Thanks for the feedback
Thomas

**Comment by tcalmant** [comment]
Hi,

After looking at the code, the order of inheritance doesn't really matter as the method overridden by the Mixin are themselves overridden by the `PooledJSONRPCServer` class.
That class uses the `process_request_thread` of the Mixin, hence the inheritance.

That being said, I'll change the order of inheritance to be coherent with what it should be.
I'm also adding time-related assertions in the server tests (we'll see how it works on Travis)
